Identifier Table

quang-ng opened the pull request:
Working on issue #16

leej3 reviewed (changes requested):
@quang-ng could you tidy this up. Rebase onto main to remove the merge commits and refactor the commits to make the git commit comments more useful: they should explain the changes you are making. It is not clear to me why tests and portions of the code are being deleted for this task.

quang-ng commented:
> @quang-ng could you tidy this up. Rebase onto main to remove the merge commits and refactor the commits to make the git commit comments more useful: they should explain the changes you are making. It is not clear to me why tests and portions of the code are being deleted for this task.

Sorry, @leej3, for any confusion caused. I removed the unit tests because I refactored the pdf_uploader to make it cleaner. Some of the unit tests were redundant, so I eliminated those.

quang-ng committed:
Add Identifier table and model definition

Add metadata conversion utility and update PDFUploader and tests
Add tests to verify identifier and document uploads, provenance records, and relationships
add test cases for upload scenarios
Fix alembic upgrade command to target all heads

leej3 reviewed (approved):
Thanks @quang-ng. No worries. Coming back from the break just made things more difficult for me to interpret. I squashed the changes to simplify the git history. The changes look good to me.
